paasta_tools/setup_marathon_job.py

```diff
@@ -306,6 +306,8 @@ def log_bounce_action(line, level='debug'):
             level='event',
         )
         return None
+    elif (not all_old_tasks) and new_app_running:
+        return None
     else:
         return 60
```

Comment:
I think this whole section might be better arranged as:

    if all_old_tasks or (not new_app_running):
        # still more work to do
        return 60

    if all([
        (apps_to_kill or tasks_to_kill),
        apps_to_kill == list(old_app_live_happy_tasks),
        tasks_to_kill == all_old_tasks,
    ]):
        log_bounce_action(
            line='%s bounce on %s finishing. Now running %s' %
            (
                bounce_method,
                serviceinstance,
                marathon_jobid
            ),
            level='event',
        )

    return None

Maybe with everything after `return 60` in an else block.